src/EmergencyMultisig.sol

```diff
@@ -10,66 +10,58 @@ import {PluginUUPSUpgradeable} from "@aragon/osx/core/plugin/PluginUUPSUpgradeab
 
 import {ProposalUpgradeable} from "@aragon/osx/core/plugin/proposal/ProposalUpgradeable.sol";
 import {Addresslist} from "@aragon/osx/plugins/utils/Addresslist.sol";
-import {IMultisig} from "./interfaces/IMultisig.sol";
+import {IEmergencyMultisig} from "./interfaces/IEmergencyMultisig.sol";
+import {OptimisticTokenVotingPlugin} from "./OptimisticTokenVotingPlugin.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+uint64 constant EMERGENCY_MULTISIG_PROPOSAL_EXPIRATION_PERIOD = 8 days;
 
 /// @title Multisig - Release 1, Build 1
-/// @author Aragon Association - 2022-2023
+/// @author Aragon Association - 2022-2024
 /// @notice The on-chain multisig governance plugin in which a proposal passes if X out of Y approvals are met.
-contract EmergencyMultisig is
-    IMultisig,
-    IMembership,
-    PluginUUPSUpgradeable,
-    ProposalUpgradeable,
-    Addresslist
-{
+contract EmergencyMultisig is IEmergencyMultisig, IMembership, PluginUUPSUpgradeable, ProposalUpgradeable {
     using SafeCastUpgradeable for uint256;
 
     /// @notice A container for proposal-related information.
     /// @param executed Whether the proposal is executed or not.
     /// @param approvals The number of approvals casted.
     /// @param parameters The proposal-specific approve settings at the time of the proposal creation.
     /// @param approvers The approves casted by the approvers.
-    /// @param actions The actions to be executed when the proposal passes.
-    /// @param _allowFailureMap A bitmap allowing the proposal to succeed, even if individual actions might revert. If the bit at index `i` is 1, the proposal succeeds even if the `i`th action reverts. A failure map value of 0 requires every action to not revert.
+    /// @param destinationActionsHash The hash of the serialized list of actions to be executed
+    /// @param encryptedPayloadURI The IPFS URI where a JSON with the encrypted payload is pinned
+    /// @param destinationPlugin The address of the plugin where the proposal will be created if it passes.
     struct Proposal {
         bool executed;
         uint16 approvals;
         ProposalParameters parameters;
         mapping(address => bool) approvers;
-        IDAO.Action[] actions;
-        uint256 allowFailureMap;
+        bytes32 destinationActionsHash;
+        bytes encryptedPayloadURI;
+        OptimisticTokenVotingPlugin destinationPlugin;
     }
 
     /// @notice A container for the proposal parameters.
     /// @param minApprovals The number of approvals required.
     /// @param snapshotBlock The number of the block prior to the proposal creation.
-    /// @param startDate The timestamp when the proposal starts.
-    /// @param endDate The timestamp when the proposal expires.
+    /// @param expirationDate The timestamp after which non-executed proposals expire.
     struct ProposalParameters {
```

Comment:
That's a very good point. However, it's conceptually the same as if the standard Multisig signers approved a proposal to remove all signers and to add new ones. No conceptual difference between Multisig and EmergencyMultisig.

For such change to happen:
- A standard Multisig proposal would need to pass, the Optimistic proposal should not be vetoed and then the DAO should apply these changes
- A supermajority of signers should approve such proposal to rule themselves out and execution would be immediate

While technically it is possible, by design, signers first need to approve themselves being kicked out

Reply:
This looks like a very specific edge case. Currently we are storing the global `minApprovals` and the per-proposal `minApprovals`. 

If we moved `minApprovals` to be a global setting then you could have proposals created when `minApprovals` was 6, eventually executed with just 1 approval because the list/setting changed in meanwhile.

As you said, you can recreate that same proposal 10 seconds later and approve it with the new signers list, and I would personally advocate for such a thing, because it makes things explicit and notorious.

Allowing a minApprovals=6 proposal to be executed with less than 6 looks like hidden magic, which I would personally discourage